services/spar/test-integration/Util/Core.hs

```diff
@@ -205,6 +209,17 @@ pendingWith :: (HasCallStack, MonadIO m) => String -> m ()
 pendingWith = liftIO . Test.Hspec.pendingWith
 
 
+-- | Run a probe a couple of times, until a "good" value materializes or until patience runs
+-- out.
+eventually :: TestSpar a -> (a -> Bool) -> TestSpar a
+eventually action good = do
+    env <- ask
+    liftIO $ retrying
+        (exponentialBackoff 50 <> limitRetries 15)
```

Comment:
Nitpick: 50µs is rather small for anything IO-related.

Reply:
i set it to 1000.